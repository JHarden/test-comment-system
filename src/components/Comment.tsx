import React from "react";

export interface CommentProps {
    author: string;
    text: string;
    date: string;
}

const commentStyle ={
    color: 'green',
    outline: '1px solid green'
}
class Comment extends React.Component<CommentProps, {}> {

    render() {
        const {author, text, date} = this.props;

        return(
            <div style={commentStyle}> 
                <div>{author}</div>
                <div>{text}</div>
                <div>{date}</div>
            </div>
        )
    }
}

export default Comment;