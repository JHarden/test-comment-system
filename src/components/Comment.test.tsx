import React from 'react';
import Comment from './Comment';
import renderer from 'react-test-renderer'
const mock = () => { }

const props = {
  author: 'john',
  text: 'this is a comment!',
  date: 'yesterday'
}

test('Comment component test', () => {
  const component = renderer.create(
    <Comment {...props}></Comment>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
