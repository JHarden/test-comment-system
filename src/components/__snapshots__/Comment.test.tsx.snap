// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component test 1`] = `
<div
  style={
    Object {
      "color": "green",
      "outline": "1px solid green",
    }
  }
>
  <div>
    john
  </div>
  <div>
    this is a comment!
  </div>
  <div>
    yesterday
  </div>
</div>
`;

exports[`RaceSelector component test 1`] = `
<div
  style={
    Object {
      "color": "green",
      "outline": "1px solid green",
    }
  }
>
  <div>
    john
  </div>
  <div>
    this is a comment!
  </div>
  <div>
    yesterday
  </div>
</div>
`;
